persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/metrics/StatisticsExporter.java

```diff
@@ -0,0 +1,277 @@
+package org.infinispan.persistence.rocksdb.metrics;
+
+public interface StatisticsExporter {
```

Comment:
I don't see an advantage to using an interface here, unless you can envisage there being a need for multiple implementations?

Reply:
Other than my old JMX knowledge.. no. I assume then that the interfaces are no longer needed.

Reply:
@danberindei Are the `@ManagedAttribute` annotations on the `StatisticsExporterImpl` all that is required?

Reply:
I assume this PR is related to the conversation that started on Apr. 9 on this zulip topic https://infinispan.zulipchat.com/#narrow/stream/118645-infinispan/topic/metrics  All mp metrics related code depends on the processing of the ```@ManagedAttribute``` at build time and the presence of an infinispan module in this maven module. I did not look in detail at this PR but I assume we got past that point and now we're improving the code.

CacheMetricsRegistration is a cache-level component that has to be started for the magic to happen by itself. It will scan the contents of the component registry of the cache and will export metrics for all components that have the annotations. This happens during cache starting event. Components that are registered after this do not get this treatment.  So if you register components manually you'll need to call CacheMetricsRegistration.registerMetrics or registerExternalMetrics. (I will not repeat here what the javadocs tell you about the difference between the two.) A sample invocation here: https://github.com/infinispan/infinispan/blob/11.0.0.Dev04/query/src/main/java/org/infinispan/query/impl/LifecycleManager.java#L195

Reply:
Btw, the way the query module exports its stats as mp metrics is probably very similar to what @burmanm intends to achieve

Reply:
A `@MBean` annotation is also required on the class. An interface is not required.